mbed_tools_ci_scripts/license_files.py

```diff
@@ -0,0 +1,85 @@
+#
+# Copyright (c) 2020-2020, Arm Mbed. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Apply copyright and licensing to all source files present in a project.
+
+This is to comply with OpenChain certification;
+https://github.com/OpenChain-Project/Curriculum/blob/master/guides/reusing_software.md#2-include-a-copyright-notice-and-license-in-each-file
+"""
+import argparse
+import logging
+import subprocess
+import sys
+import tempfile
+from datetime import datetime
+from mbed_tools_ci_scripts.utils.configuration import configuration, ConfigurationVariable
+from mbed_tools_ci_scripts.utils.logging import set_log_level, log_exception
+from pathlib import Path
+
+logger = logging.getLogger(__name__)
+
+LICENCE_HEADER_TEMPLATE = """Copyright (c){date}, {author}. All rights reserved.
+SPDX-License-Identifier: {licence_identifier}
+"""
+
+
+class CopyrightNotifierToolWrapper:
+    """Wrapper over the licenseheaders tool.
+
+    See https://github.com/johann-petrak/licenseheader for more details about the tool in use.
+    copyright (https://github.com/knipknap/copyright) was first considered but
+    comprises quite a few bugs and does not seem active anymore.
+    """
+
+    def run(self, verbose_count: int) -> None:
+        """Puts a copyright notice at the top of every source file."""
+        template_string = LICENCE_HEADER_TEMPLATE.format(
+            licence_identifier=configuration.get_value(ConfigurationVariable.FILE_LICENCE_IDENTIFIER),
+            author="${owner}",
+            date=" ${years}",
+        )
+        with tempfile.NamedTemporaryFile(suffix=".tmpl", delete=False) as template_file:
+            template_file_path = Path(template_file.name)
+            logger.debug(f"Creates template file in {str(template_file_path)}")
+            template_file.write(template_string.encode("utf8"))
+            template_file.close()
+            copyright_config = self.get_tool_config(template_file_path)
+            self.run_licensehearders(copyright_config, verbose_count)
+
+    def run_licensehearders(self, config: dict, verbose_count: int) -> None:
+        """Runs licenseheaders tool."""
+        args = ["licenseheaders"]
+        args.extend([item for pair in {f"--{k}": str(v) for (k, v) in config.items()}.items() for item in pair])
```

Comment:
Any way to simplify this expression? It's a bit complicated. Maybe just a regular `for` loop would be clearer?

Reply:
I broke it down